services/channel/service/migrations/committed/000003-added-initial-tables.sql

```diff
@@ -0,0 +1,892 @@
+-- |_|  |_|\__/_| \_. ||_||_|/__/ \__|
+--                |__/ 
+--------------------------------------
+
+-- table: playlists
+DROP TABLE IF EXISTS app_public.playlists CASCADE;
+CREATE TABLE app_public.playlists(
+  id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
+  channel_id UUID NOT NULL REFERENCES app_public.channels(id) ON DELETE CASCADE,
+  start_date_time TIMESTAMPTZ NOT NULL,
+  calculated_duration_in_seconds NUMERIC(13, 5) NOT NULL,
+  published_date TIMESTAMPTZ,
+  published_user TEXT
+);
+
+SELECT ax_define.define_audit_date_fields_on_table('playlists', 'app_public');
+SELECT ax_define.define_audit_user_fields_on_table('playlists', 'app_public', ':DEFAULT_USERNAME');
+SELECT ax_define.define_authentication('ADMIN,CHANNELS_EDIT,CHANNELS_VIEW', 'ADMIN,CHANNELS_EDIT', 'playlists', 'app_public');
+
+SELECT ax_define.set_enum_as_column_type('publication_state', 'playlists', 'app_public', 'publication_state', 'app_public', 'NOT_PUBLISHED');
+SELECT ax_define.set_enum_domain('publication_state', 'playlists', 'app_public', 'publication_state_enum', 'app_public');
+SELECT app_hidden.define_publish_trigger('playlists', 'app_public', 'start_date_time,calculated_duration_in_seconds,channel_id');
+SELECT ax_define.define_subscription_triggers('id', 'playlists', 'app_public', 'playlists', 'PLAYLIST');
+
+SELECT ax_define.define_indexes_with_id('start_date_time', 'playlists', 'app_public');
+SELECT ax_define.define_indexes_with_id('created_date', 'playlists', 'app_public');
+SELECT ax_define.define_indexes_with_id('updated_date', 'playlists', 'app_public');
+SELECT ax_define.define_indexes_with_id('published_date', 'playlists', 'app_public');
+SELECT ax_define.define_index('channel_id', 'playlists', 'app_public');
+
+GRANT SELECT, DELETE ON app_public.playlists TO ":DATABASE_GQL_ROLE";
+GRANT INSERT (channel_id, start_date_time, calculated_duration_in_seconds) ON app_public.playlists TO ":DATABASE_GQL_ROLE";
+GRANT UPDATE (start_date_time, calculated_duration_in_seconds) ON app_public.playlists TO ":DATABASE_GQL_ROLE";
+
+DROP FUNCTION IF EXISTS app_public.playlists_calculated_end_date_time;
+CREATE FUNCTION app_public.playlists_calculated_end_date_time(playlists playlists) RETURNS TIMESTAMPTZ AS $$
+  SELECT playlists.start_date_time + playlists.calculated_duration_in_seconds * INTERVAL '1 seconds'
+$$ LANGUAGE sql STABLE;
+
+-------------- #program ------------------
+--  ___                             
+-- | _ \ _ _  ___  __ _  _ _  __ _  _ __  
+-- |  _/| '_|/ _ \/ _` || '_|/ _` || '  \ 
+-- |_|  |_|  \___/\__. ||_|  \__/_||_|_|_|
+--                |___/           
+------------------------------------------
+
+DROP TABLE IF EXISTS app_public.programs CASCADE;
+CREATE TABLE app_public.programs(
+  id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
+  playlist_id UUID NOT NULL REFERENCES app_public.playlists(id) ON DELETE CASCADE,
+  sort_index INT NOT NULL,
+  title TEXT NOT NULL,
+  image_id UUID,
+  video_id UUID NOT NULL,
```

Comment:
I'm not sure if `video_id` should be kept at program level since
we now have an easier access to the parent entity. (Not sure about `image_id`)

Here's an example:
- Assign some video to the movie that does not match the channel placeholder
  video, and assign said movie as a program item
- Try to publish the playlist - this error will be shown `The placeholder video and playlist video(s) must each have at least one stream with the same resolution, frame rate, and bitrate.`
- It felt logical to go to movie details, and assign a different video. But this
  would not remove the error above if you try to publish the playlist again. You
  would need to go to the playlist, unasign related movie, save changes, and
  assign it again - this would update the stored video_id, which would make the
  validation use updated `video_id`

Or maybe we need some events in the media service, e.g. image/video changed -
which channel service would pick up and update relations for its program items?

Reply:
Future improvement is fine with me,